cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/worker/DefaultWorkerConfigurer.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.worker;
+
+import co.cask.cdap.api.Resources;
+import co.cask.cdap.api.worker.Worker;
+import co.cask.cdap.api.worker.WorkerConfigurer;
+import co.cask.cdap.api.worker.WorkerSpecification;
+import co.cask.cdap.internal.lang.Reflections;
+import co.cask.cdap.internal.specification.PropertyFieldExtractor;
+import com.clearspring.analytics.util.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.google.common.reflect.TypeToken;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Default implementation of the {@link WorkerConfigurer}.
+ */
+public class DefaultWorkerConfigurer implements WorkerConfigurer {
+
+  private final String className;
+  private final Map<String, String> propertyFields;
+
+  private String name;
+  private String description;
+  private Resources resource;
+  private int instances;
+  private Map<String, String> properties;
+  private Set<String> datasets;
+
+  public DefaultWorkerConfigurer(Worker worker) {
+    this.name = worker.getClass().getSimpleName();
```

Comment:
Probably also want to have a `setName()` method to allow setting the name inside `configure`. Currently we allow that by having `addWorker(String name, Worker worker)` inside the Service, but also good to have that here.